### DISCUSSION TITLE
NAU7802 Library Conversion from Circuitpython/Arduino

### DISCUSSION THREAD

**Discussion by balbert1** [micropython/micropython#13472] [Libraries & Drivers]
I recently bought Adafruit's NAU7802 and I've started trying to implement it with a sensor on my RP Pico with micropython. I'm having more trouble than I thought I would with converting existing libraries to work with Micropython. There are currently libraries for both Arduino (https://github.com/adafruit/Adafruit_NAU7802) and Circuitpython (https://github.com/adafruit/CircuitPython_NAU7802) but I've realized I'm not proficient enough with either of those to understand where they diverge and how I should go about translating them.

I'm wondering if anyone has any tips or ideas on how to get started with converting either of those and which would be easier to translate? I'm also wondering if making a simple library from scratch would be too much for someone like me who does not have much experience with I2C devices like this one. Thanks!

**Comment by ElTopollillo1990** [comment]
Have you looked at using Adafruit's CircuitPython's library within MicroPython using BLINKA ? -> https://learn.adafruit.com/circuitpython-libraries-on-micropython-using-the-raspberry-pi-pico

I haven't used it yet but is on my near-term plans as I want to stick with MP but would like to use some CP libraries so as to no re-invent the wheel.

**Comment by jjahn504** [comment]
I made a code. You can download a example at the below link.
 https://github.com/jjahn504/pico_NAU7802_micropython
